packages/@webex/plugin-cc/src/cc.ts

```diff
@@ -80,12 +88,20 @@ export default class ContactCenter extends WebexPlugin implements IContactCenter
     this.webSocket.on(EVENT, this.processEvent);
   }
 
-  private processEvent = (event: WebSocketEvent): void => {
+  private processEvent = async (event: WebSocketEvent): Promise<void> => {
     switch (event.type) {
-      case CC_EVENTS.WELCOME:
-        this.ciUserId = event.data.agentId;
-        this.handleEvent(REGISTER_EVENT, `Success: CI User ID is ${this.ciUserId}`); // TODO: Will send AgentPRofile object as part of Parv's PR
+      case CC_EVENTS.WELCOME: {
+        const agentId = event.data.agentId;
+        const agentConfig = new AgentConfig(agentId, this.$webex, this.wccApiUrl);
+        this.agentConfig = await agentConfig.getAgentProfile();
+        this.$webex.logger.log(
+          `agent config is: ${JSON.stringify(this.agentConfig)} file: ${CC_FILE} method: ${
+            this.register.name
+          }`
+        );
+        this.handleEvent(REGISTER_EVENT, `Success: Agent Profile is ${this.agentConfig}`);
```

Comment:
We shouldn't be returning just string data here. Agent Config should be returned as payload, and we require app.js changes to receive this information and reflect it in samples app

Reply:
Code Updated!